dev/customer_testing/lib/customer_test.dart

```diff
@@ -106,7 +110,60 @@ class CustomerTest {
 
   final List<String> contacts;
   final List<String> fetch;
+  final List<String> setup;
   final List<Directory> update;
   final List<String> tests;
   final int? iterations;
+
+  static void _maybeAddTestConfig(
+    String line, {
+    required _TestDirective directive,
+    required List<String> directiveValues,
+    bool platformAgnostic = true,
+  }) {
+    final List<_PlatformType> platforms = platformAgnostic
+        ? <_PlatformType>[_PlatformType.all]
+        : _PlatformType.values;
+    for (final _PlatformType platform in platforms) {
+      final String directiveName = _directive(directive, platform: platform);
+      if (line.startsWith(directiveName) && platform.conditionMet) {
+        directiveValues.add(line.substring(directiveName.length));
+      }
+    }
+  }
+
+  static String _directive(
+    _TestDirective directive, {
+    _PlatformType platform = _PlatformType.all,
+  }) {
+    return switch (platform) {
+      _PlatformType.all => '${directive.name}=',
+      _ => '${directive.name}.${platform.name}=',
+    };
+  }
+}
+
+enum _PlatformType {
+  all,
+  windows,
+  macos,
+  linux,
+  posix;
+
+  bool get conditionMet => switch (this) {
+        _PlatformType.all => true,
+        _PlatformType.windows => Platform.isWindows,
+        _PlatformType.macos => Platform.isMacOS,
+        _PlatformType.linux => Platform.isLinux,
+        _PlatformType.posix => Platform.isLinux || Platform.isMacOS,
+      };
+}
+
+enum _TestDirective {
+  contact,
+  fetch,
+  setup,
```

Comment:
Can you please also update https://github.com/flutter/tests/blob/main/registry/template.test to mention and document the new directive? Thank you!

Reply:
Done in this PR https://github.com/flutter/tests/pull/267